app/src/main/java/com/github/swent/echo/data/repository/SimpleRepository.kt

```diff
@@ -0,0 +1,112 @@
+ */
+class SimpleRepository(authenticationService: AuthenticationService) : Repository {
+
+    private val associations = mutableSetOf<Association>()
+    private val events = mutableListOf<Event>()
+    private val tags =
+        mutableSetOf(
+            Tag("1", "Sport", Repository.ROOT_TAG_ID),
+            Tag("2", "Culture", Repository.ROOT_TAG_ID),
+            Tag("3", "Music", Repository.ROOT_TAG_ID),
+            Tag("4", "Football", "1"),
+            Tag("5", "Basketball", "1"),
+            Tag("6", "Theatre", "2"),
+        )
+    private val userProfiles = mutableSetOf<UserProfile>()
+
+    init {
+        // Populate the repository with sample data
+        SAMPLE_EVENTS.forEach { event ->
+            events.add(event)
+            tags.addAll(event.tags)
+            if (event.organizer != null) {
+                associations.add(event.organizer)
+            }
+            userProfiles.add(
+                UserProfile(
+                    event.creator.userId,
+                    event.creator.name,
+                    SemesterEPFL.BA3,
+                    SectionEPFL.IN,
+                    emptySet(),
+                )
+            )
+        }
+
+        // Add a user profile for the current user
+        val userId = authenticationService.getCurrentUserID()
+        if (userId != null) {
+            userProfiles.add(
+                UserProfile(
+                    userId,
+                    "John Doe",
+                    SemesterEPFL.BA3,
+                    SectionEPFL.IN,
+                    tags.take(2).toSet(),
+                )
+            )
+        }
+    }
+
+    override suspend fun getAssociation(associationId: String): Association {
+        return associations.find { it.associationId == associationId }!!
+    }
+
+    override suspend fun getAllAssociations(): List<Association> {
+        return associations.toList()
+    }
+
+    override suspend fun getEvent(eventId: String): Event {
+        return events.find { it.eventId == eventId }!!
+    }
+
+    override suspend fun createEvent(event: Event): String {
+        val eventId = events.hashCode().toString()
+        events.add(event.copy(eventId = eventId))
+        return eventId
+    }
+
+    override suspend fun setEvent(event: Event) {
+        events.add(event)
+    }
+
+    override suspend fun getAllEvents(): List<Event> {
+        return events
+    }
+
+    override suspend fun getTag(tagId: String): Tag {
+        return tags.find { it.tagId == tagId }!!
+    }
+
+    override suspend fun getAllTags(): List<Tag> {
+        return tags.toList()
+    }
+
+    override suspend fun getSubTags(tagId: String): List<Tag> {
+        return tags.filter { it.parentId == tagId }
+    }
+
+    override suspend fun getUserProfile(userId: String): UserProfile? {
+        return userProfiles.find { it.userId == userId }
+    }
+
+    override suspend fun setUserProfile(userProfile: UserProfile) {
+        userProfiles.add(userProfile)
```

Comment:
Same problem as setEvent, it should first try to find an existing userProfile (same userId)